lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -74,6 +73,23 @@ def relevant_included_associations_for(serializer, nested_associations = {})
           end
         end
 
+        def add_relationship_for(parent, key, serializer, opts)
+          # mutates parent
+          apply_relationship_to(parent, key) do
+            if serializer.respond_to?(:each)
+              to_many_relationship_for(serializer, opts)
+            else
+              to_one_relationship_for(serializer, opts)
+            end
+          end
+
+          parent
+        end
```

Comment:
I'm down with that idea. I'm unsure of where the to_many_relationship_for and to_one_relationship_for would go though. 

like, would each adapter have an "array" version of itself?, and then we could just call the method `relationship_for`